Problems with extend BannerComponent

**Describe the bug**
Observed here: https://stackoverflow.com/questions/71030337/not-able-to-use-cx-media-component-with-a-media-container
I can see TS compilation errors while try extends BannerComponent. It's related to used wrong types:
- Type 'string | undefined' is not assignable to type 'string | any[]'. `[url]="data.urlLink"`
- Type 'string | null' is not assignable to type 'string' `[target]="getTarget(data)"`
- Type 'CmsBannerComponentMedia | CmsResponsiveBannerComponentMedia | undefined' is not assignable to type 'MediaContainer'.
  Type 'undefined' is not assignable to type 'MediaContainer'.  ` [container]="data.media"`

**Tell us the version of Spartacus**
- Library version: 4.3

**To Reproduce**
Steps to reproduce the behavior:

1 Create a new Angular app, add Spartacus by schematics, set any possible backend
2 Add own MyBannerComponent extends BannerComponent, use copied template from BannerComponent
3 Try to compile
4 See error list:
```
Error: src/app/my-banner/my-banner.component.html:3:4 - error TS2322: Type 'string | undefined' is not assignable to type 'string | any[]'.
  Type 'undefined' is not assignable to type 'string | any[]'.

3   [url]="data.urlLink"
     ~~~

  src/app/my-banner/my-banner.component.ts:6:16
    6   templateUrl: './my-banner.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component MyBannerComponent.


Error: src/app/my-banner/my-banner.component.html:4:4 - error TS2322: Type 'string | null' is not assignable to type 'string'.
  Type 'null' is not assignable to type 'string'.

4   [target]="getTarget(data)"
     ~~~~~~

  src/app/my-banner/my-banner.component.ts:6:16
    6   templateUrl: './my-banner.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component MyBannerComponent.


Error: src/app/my-banner/my-banner.component.html:7:14 - error TS2322: Type 'CmsBannerComponentMedia | CmsResponsiveBannerComponentMedia | undefined' is not assignable to type 'MediaContainer'.
  Type 'undefined' is not assignable to type 'MediaContainer'.

7   <cx-media [container]="data.media"></cx-media>
               ~~~~~~~~~

  src/app/my-banner/my-banner.component.ts:6:16
    6   templateUrl: './my-banner.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component MyBannerComponent.

```

**Expected behavior**
The component should compile properly, without any input.

**Desktop (please complete the following information):**
- OS: Windows 10
- Browser: Chrome

**Additional context**
- Analogical issue for the different component: https://github.com/SAP/spartacus/issues/13652

--- a/projects/storefrontlib/cms-components/content/banner/banner.component.html
+++ b/projects/storefrontlib/cms-components/content/banner/banner.component.html
@@ -1,9 +1,11 @@
-<cx-generic-link
-  *ngIf="data$ | async as data"
-  [url]="data.urlLink"
-  [target]="getTarget(data)"
->
-  <p class="headline" *ngIf="data.headline" [innerHTML]="data.headline"></p>
-  <cx-media [container]="data.media"></cx-media>
-  <p class="content" *ngIf="data.content" [innerHTML]="data.content"></p>
-</cx-generic-link>
+<ng-container *ngIf="data$ | async as data">
+  <cx-generic-link
+    *ngIf="data.urlLink"
+    [url]="data.urlLink"
+    [target]="getTarget(data)"
+  >
+    <p class="headline" *ngIf="data.headline" [innerHTML]="data.headline"></p>
+    <cx-media [container]="getImage(data)"></cx-media>
+    <p class="content" *ngIf="data.content" [innerHTML]="data.content"></p>
+  </cx-generic-link>
+</ng-container>

--- a/projects/storefrontlib/cms-components/content/banner/banner.component.ts
+++ b/projects/storefrontlib/cms-components/content/banner/banner.component.ts
@@ -1,5 +1,5 @@
 import { ChangeDetectionStrategy, Component, HostBinding } from '@angular/core';
-import { CmsBannerComponent } from '@spartacus/core';
+import { CmsBannerComponent, Image, ImageGroup } from '@spartacus/core';
 import { Observable } from 'rxjs';
 import { tap } from 'rxjs/operators';
 import { CmsComponentData } from '../../../cms-structure/page/model/cms-component-data';
@@ -10,7 +10,7 @@ import { CmsComponentData } from '../../../cms-structure/page/model/cms-componen
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class BannerComponent {
-  @HostBinding('class') styleClasses: string;
+  @HostBinding('class') styleClasses: string | undefined;
 
   data$: Observable<CmsBannerComponent> = this.component.data$.pipe(
     tap((data) => (this.styleClasses = data.styleClasses))
@@ -24,5 +24,15 @@ export class BannerComponent {
    */
   getTarget(data: CmsBannerComponent): string | null {
     return data.external === 'true' || data.external === true ? '_blank' : null;
+  }
+
+  getImage(data: CmsBannerComponent): Image | ImageGroup | undefined {
+    if (data.media) {
+      if ('url' in data.media) {
+        return data.media as Image;
+      } else {
+        return data.media as ImageGroup;
+      }
+    }
   }
 }

--- a/projects/storefrontlib/shared/components/generic-link/generic-link.component.ts
+++ b/projects/storefrontlib/shared/components/generic-link/generic-link.component.ts
@@ -68,7 +68,7 @@ export class GenericLinkComponent implements OnChanges {
   private routeParts: RouteParts = {};
 
   @Input() url: string | any[];
-  @Input() target: string;
+  @Input() target: string | null;
   @Input() id: string;
   @Input() class: string;
   @Input() style: string;

